### ISSUE TITLE
Rule idea: flag attempts to reuse animated style?

### ISSUE THREAD

**Issue by helixbass** [wcandillon/eslint-plugin-reanimated#19]
Per https://github.com/software-mansion/react-native-reanimated/issues/1363 (which I also ran into), you cannot (at least currently) reuse an animated style on multiple elements

So might be nice to have a rule that catches this at lint time?

Eg this should get flagged:
```
const style = useAnimatedStyle(() => ({
  opacity: opacity.value
}))

return (
  <>
    <Animated.View style={style} />
    <Animated.View style={style} />
  </>
)
```

I might be able to try and implement this rule, but not sure what the most robust strategy is for detecting the reuse? Eg if you can use the scope manager's references to track multiple references to the same style object that are usages in `style` props? And then for knowing that it's an animated style object, whether to eg just rely on the Typescript type being what you'd expect for an animated style (`Animated.AnimatedStyle<...>`) vs manually inspecting that it's defined via `useAnimatedStyle()`?

**Comment by wcandillon** [comment]
haha that's an awesome idea, thanks Julian :)
